mne/io/edf/edf.py

```diff
@@ -380,6 +380,17 @@ def _get_edf_info(fname, stim_channel, annot, annotmap, eog, misc, exclude,
         fid.read(32 * nchan).decode()  # reserved
         assert fid.tell() == header_nbytes
 
+        if n_records == -1:
+            warn('Unable to read the number of records from the header '
+                 '(perhaps the recording was not stopped before exiting). '
+                 'Inferring from the file size.')
+            fid.seek(0, 2)
+            n_bytes = fid.tell()
+            n_data_bytes = n_bytes - header_nbytes
+            total_samps = (n_data_bytes // 3 if subtype == '24BIT'
+                           else n_data_bytes // 2)
+            edf_info['n_records'] = n_records = total_samps // np.sum(n_samps)
```

Comment:
It probably makes sense to just always do this calculation, and `warn` if `n_records` is missing or there is a mismatch between expected and calculated